File: app/qr.py
"""Geração de QR Code para ingressos (equivalente ao QRCodeGenerator.java)."""
import base64
from io import BytesIO

import qrcode


def generate_qr_code_base64(data: str) -> str:
    img = qrcode.make(data, box_size=8, border=2)
    buffer = BytesIO()
    img.save(buffer, format="PNG")
    return base64.b64encode(buffer.getvalue()).decode("utf-8")

File: src/utils/qr.js
'use strict';
/**
 * Geração de QR Code para ingressos (equivalente a app/qr.py).
 */
const QRCode = require('qrcode');

/**
 * Gera um QR code em PNG codificado como base64 (sem o prefixo data URI),
 * igual ao retorno de generate_qr_code_base64 no backend Python.
 * @param {string} data
 * @returns {Promise<string>}
 */
async function generateQrCodeBase64(data) {
  const buffer = await QRCode.toBuffer(data, {
    type: 'png',
    width: 8 * 37, // aproxima o box_size=8 usado no qrcode Python
    margin: 2,
  });
  return buffer.toString('base64');
}

module.exports = { generateQrCodeBase64 };
